File: node/fundamentals/closure/closure.js

import { functionToCall } from './moduleToImport.mjs';

function mainFunction() {
    
    let a = 5;

    function printThis() {
        console.log(a);
        // console.log(b);
        let c = 7;
    }

    // console.log(c);
    functionToCall(printThis);
}

mainFunction();

File: python/fundamentals/closure/closure.py
import pathlib, sys
sys.path.append(str(pathlib.Path(__file__).resolve().parents[0]))
from moduleToImport import functionToCall


def mainFunction(arrayOfArguments):

    a = 5

    def printThis():
        print(a)
        # print(b)
        c = 7

    # print(c)
    functionToCall(printThis)

# mainFunction(sys.argv)

